src/services/review/RepoCheckerService.ts

```diff
@@ -0,0 +1,223 @@
+import fs from "fs"
+import path from "path"
+
+import jsdom from "jsdom"
+import { forEach } from "lodash"
+import { marked } from "marked"
+import { ResultAsync, errAsync, ok } from "neverthrow"
+
+import { EFS_VOL_PATH_STAGING } from "@root/constants"
+import SiteCheckerError from "@root/errors/SiteCheckerError"
+import { RepoError } from "@root/types/siteChecker"
+import { ALLOWED_FILE_EXTENSIONS } from "@root/utils/file-upload-utils"
+
+const { JSDOM } = jsdom
+
+export default class RepoCheckerService {
+  getAllRepos(): ResultAsync<string[], Error> {
+    // todo
+    return errAsync(new Error("Not implemented"))
+  }
+
+  runCheckerForAllRepos(): ResultAsync<undefined, Error> {
+    // todo
+    return errAsync(new Error("Not implemented"))
+  }
+
+  async checkRepo(repo: string) {
+    const repoPath = path.join(EFS_VOL_PATH_STAGING, repo)
+
+    const mdFiles = new Set<string>()
+    // get list of all documents
+    const checker = this.getListOfMarkdownFiles(repoPath)
+      .andThen((files) => {
+        files.forEach((file) => {
+          mdFiles.add(file)
+        })
+        return this.getAllMediaPath(repoPath)
+      })
+      .andThen((setOfAllMediaPath) =>
+        ResultAsync.fromPromise(
+          this.getAllPermalinks(repoPath, mdFiles),
+          (error) => new SiteCheckerError(`${error}`)
+        ).andThen((setOfAllPermalinks) =>
+          ok(new Set([...setOfAllMediaPath, ...setOfAllPermalinks]))
+        )
+      )
+      .andThen((setOfAllMediaAndPagesPath) => {
+        mdFiles.forEach(async (fileName) => {
+          const file = fs.readFileSync(path.join(repoPath, fileName), "utf8")
+          const filePermalink = file
+            .split("---")[1]
+            .split("permalink: ")[1]
+            .split("\n")[0]
+          const fileContent = file.split("---")[2]
+          if (!fileContent) {
+            return
+          }
+          const html = await marked.parse(fileContent)
+          const dom = new JSDOM(html)
+          const anchorTags = dom.window.document.querySelectorAll("a")
+          forEach(anchorTags, (tag) => {
+            const href = tag.getAttribute("href")
+            const text = tag.textContent
+            if (href) {
+              const encodedHref = this.documentPathDecoder(href)
```

Comment:
Hm 🤔 , the var name is `encodedHref`, but conditionally `documentPathDecoder()` calls `decodeURIComponent()`, so the result is a decoded href?. In fact based on this code, if `href` is a full url, `documentPathDecoder()` should use `decodeUri()`, rather than `decodeURIComponent()`, no? 🤔

Reply:
213feca5939bbbcc3ff0453d7496f22c347b5ed0
fair point